pkgs/development/web/bun/default.nix

```
@@ -64,4 +64,39 @@ stdenvNoCC.mkDerivation rec {
     maintainers = with maintainers; [ DAlperin jk ];
     platforms = builtins.attrNames dists;
   };
+  passthru = {
+    sources = {
+        "aarch64-darwin" = fetchurl {
+            url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-darwin-aarch64.zip";
+            sha256 = "sha256-R17Ga4C6PSxfL1bz6IBjx0dYFmX93i0y8uqxG1eZKd4=";
+        }; 
+        "aarch64-linux" = {
+            url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-linux-aarch64.zip";
+            sha256 = "sha256-KSC4gdsJZJoPjMEs+VigVuqlUDhg4sL054WRlAbB+eA=";
+        };
+        "x86_64-darwin" = {
+            url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-darwin-x64.zip";
+            sha256 = "sha256-CVqFPvZScNTudE2wgUctwGDgTyaMeN8dUNmLatcKo5M=";
+        };
+        "x86_64-linux" = { 
+            url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-linux-x64.zip";
+            sha256 = "sha256-N3hGPyp9wvb7jjpaFLJcdNIRyLvegjAe+MiV2aMS1nE=";
+        };
+    };
+    updateScript = writeShellScript "update-bun" ''
+      set -o errexit
+      export PATH="${lib.makeBinPath [ curl jq common-updater-scripts ]}"
+      NEW_VERSION=$(curl -s https://api.github.com/repos/oven-sh/bun/releases | jq ".[1].tag_name" --raw-output)
```

Or maybe even nicer:

```suggestion
      NEW_VERSION=$(curl --silent https://api.github.com/repos/oven-sh/bun/releases/latest | jq '.tag_name | ltrimstr("bun-v")' --raw-output)
```